File: app/src/main/java/com/neocaptainnemo/cv/model/Project.java
package com.neocaptainnemo.cv.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.firebase.database.IgnoreExtraProperties;
import com.google.firebase.database.PropertyName;

@IgnoreExtraProperties
public class Project implements Parcelable {

    public static final String PLATFORM_ANDROID = "android";

    public static final String PLATFORM_IOS = "ios";
    public static final Creator<Project> CREATOR = new Creator<Project>() {
        @Override
        public Project createFromParcel(Parcel in) {
            return new Project(in);
        }

        @Override
        public Project[] newArray(int size) {
            return new Project[size];
        }
    };
    @PropertyName("name")
    public String name;
    @PropertyName("description")
    public String description;
    @PropertyName("web_pic")
    public String webPic;
    @PropertyName("platform")
    public String platform;
    @PropertyName("vendor")
    public String vendor;
    @PropertyName("cover_pic")
    public String coverPic;
    @PropertyName("stack")
    public String stack;
    @PropertyName("duties")
    public String duties;
    @PropertyName("store_url")
    public String storeUrl;
    @PropertyName("git_hub")
    public String gitHub;
    @PropertyName("description_key")
    public String descriptionKey;
    @PropertyName("duties_key")
    public String dutiesKey;
    @PropertyName("name_key")
    public String nameKey;


    public Project() {
        //do nothing
    }

    protected Project(Parcel in) {
        name = in.readString();
        description = in.readString();
        webPic = in.readString();
        platform = in.readString();
        vendor = in.readString();
        coverPic = in.readString();
        stack = in.readString();
        duties = in.readString();
        storeUrl = in.readString();
        gitHub = in.readString();
        descriptionKey = in.readString();
        dutiesKey = in.readString();
        nameKey = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(name);
        parcel.writeString(description);
        parcel.writeString(webPic);
        parcel.writeString(platform);
        parcel.writeString(vendor);
        parcel.writeString(coverPic);
        parcel.writeString(stack);
        parcel.writeString(duties);
        parcel.writeString(storeUrl);
        parcel.writeString(gitHub);
        parcel.writeString(descriptionKey);
        parcel.writeString(dutiesKey);
        parcel.writeString(nameKey);
    }
}

File: app/src/main/java/com/neocaptainnemo/cv/model/Project.kt
package com.neocaptainnemo.cv.model

import android.os.Parcelable
import com.google.firebase.database.PropertyName
import com.neocaptainnemo.cv.ui.adapter.AdapterItem
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Project(
        var name: String? = null,
        var description: String? = null,
        @get: PropertyName("web_pic")
        @set: PropertyName("web_pic")
        var webPic: String? = null,
        @get: PropertyName("platform")
        @set: PropertyName("platform")
        var platform: String? = null,
        @get: PropertyName("vendor")
        @set: PropertyName("vendor")
        var vendor: String? = null,
        @get: PropertyName("cover_pic")
        @set: PropertyName("cover_pic")
        var coverPic: String? = null,
        @get: PropertyName("stack")
        @set: PropertyName("stack")
        var stack: String? = null,
        var duties: String? = null,
        @get: PropertyName("store_url")
        @set: PropertyName("store_url")
        var storeUrl: String? = null,
        @get: PropertyName("git_hub")
        @set: PropertyName("git_hub")
        var gitHub: String? = null,
        @get: PropertyName("description_key")
        @set: PropertyName("description_key")
        var descriptionKey: String? = null,
        @get: PropertyName("duties_key")
        @set: PropertyName("duties_key")
        var dutiesKey: String? = null,
        @get: PropertyName("name_key")
        @set: PropertyName("name_key")
        var nameKey: String? = null
) : Parcelable, AdapterItem {

    override val uniqueTag: String = "Project$nameKey"


    companion object {

        const val PLATFORM_ANDROID = "android"

        const val PLATFORM_IOS = "ios"
    }
}
